feature_engine/categorical_encoders.py

```diff
@@ -10,6 +10,8 @@
 from feature_engine.dataframe_checks import _is_dataframe, _check_input_matches_training_df, _check_contains_na
```

Comment:
could we move this transformer to its own python script? 

I was advised that current code format of Feature-engine is not best of practices as it contains more than 500 lines of code on each script.

Please move the new transformer to a new script called decision_tree_encoders.py

Reply:
In my opinion, we should keep the encoder in the categorical encoder for the time being to keep the API structure intact. During a future refactor, we can move the individual encoders to separate modules under packages.
for eg :
```
categorical_encoders/
  __init__.py
  decision_tree.py
  count_frequency.py
  ...
discretisers/
  __init__.py
  equal_frequency.py
  equal_width.py
  ...
```
Or if you feel DecisionTreeCategoricalEncoder being in a separate module outside of categorical_encoders is more appropriate, that would work too. Let me know your thoughts on this.

Reply:
makes sense. Thank you